proxy/envoy/discovery.go

```diff
@@ -471,19 +474,23 @@ func (ds *DiscoveryService) ListClusters(request *restful.Request, response *res
 		// set connect timeout
 		clusters.setTimeout(ds.MeshConfig.ConnectTimeout)
 
-		// apply custom policies for HTTP clusters
-		for _, cluster := range clusters {
-			insertDestinationPolicy(ds.Config, cluster)
-		}
-
-		switch auth := ds.MeshConfig.AuthPolicy; auth {
-		case config.ProxyMeshConfig_NONE:
-		case config.ProxyMeshConfig_MUTUAL_TLS:
-			// apply SSL context to enable mutual TLS between Envoy proxies
+		// egress proxy clusters reference external destinations
+		if node != egressNode {
+			// apply custom policies for HTTP clusters
 			for _, cluster := range clusters {
-				ports := model.PortList{cluster.port}.GetNames()
-				serviceAccounts := ds.Accounts.GetIstioServiceAccounts(cluster.hostname, ports)
-				cluster.SSLContext = buildClusterSSLContext(ds.MeshConfig.AuthCertsPath, serviceAccounts)
+				insertDestinationPolicy(ds.Config, cluster)
+			}
+
+			// apply auth policies
+			switch auth := ds.MeshConfig.AuthPolicy; auth {
```

Comment:
nit: just `switch ds.MeshConfig.AuthPolicy` (previously assigned it to a var since it's used in the default clause for logging).